hedera-node/src/main/java/com/hedera/services/txns/schedule/ScheduleProcessing.java

```diff
@@ -0,0 +1,288 @@
+import com.hedera.services.utils.accessors.SignedTxnAccessor;
+import com.hedera.services.utils.accessors.TxnAccessor;
+import com.hederahashgraph.api.proto.java.ResponseCodeEnum;
+import com.hederahashgraph.api.proto.java.ScheduleID;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.eclipse.collections.impl.set.mutable.primitive.LongHashSet;
+
+import static com.hedera.services.utils.EntityNum.fromScheduleId;
+import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.OK;
+import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.SCHEDULE_FUTURE_GAS_LIMIT_EXCEEDED;
+import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.SCHEDULE_FUTURE_THROTTLE_EXCEEDED;
+
+/**
+ * Class that encapsulates some of the more complex processing of scheduled transactions.
+ */
+@Singleton
+public class ScheduleProcessing {
+	private static final Logger log = LogManager.getLogger(ScheduleProcessing.class);
+
+	private final SigImpactHistorian sigImpactHistorian;
+	private final ScheduleStore store;
+	private final ScheduleExecutor scheduleExecutor;
+	private final GlobalDynamicProperties dynamicProperties;
+	private final TimedFunctionalityThrottling scheduleThrottling;
+	private final Supplier<MerkleScheduledTransactions> schedules;
+	private final long maxProcessingLoopIterations;
+
+	SigMapScheduleClassifier classifier = new SigMapScheduleClassifier();
+	SignatoryUtils.ScheduledSigningsWitness signingsWitness = SignatoryUtils::witnessScoped;
+	Predicate<ScheduleVirtualValue> isFullySigned;
+
+	@Inject
+	public ScheduleProcessing(final SigImpactHistorian sigImpactHistorian, final ScheduleStore store,
+			final ScheduleExecutor scheduleExecutor, final GlobalDynamicProperties dynamicProperties,
+			final ScheduleSigsVerifier scheduleSigsVerifier,
+			@ScheduleThrottle final TimedFunctionalityThrottling scheduleThrottling,
+			final Supplier<MerkleScheduledTransactions> schedules) {
+		this.sigImpactHistorian = sigImpactHistorian;
+		this.store = store;
+		this.scheduleExecutor = scheduleExecutor;
+		this.dynamicProperties = dynamicProperties;
+		this.scheduleThrottling = scheduleThrottling;
+		isFullySigned = scheduleSigsVerifier::areAllKeysActive;
+		this.schedules = schedules;
+		this.maxProcessingLoopIterations = dynamicProperties.schedulingMaxTxnPerSecond() * 2;
+	}
+
+	/**
+	 * Expires all scheduled transactions, that are in a final state, having an expiry before consensusTime,
+	 * and having an expiry before any transaction that is ready to execute.
+	 */
+	public void expire(Instant consensusTime) {
+
+		// we _really_ don't want to loop forever. If the database isn't working, then it's possible. So we put
+		// an upper limit on the number of iterations here.
+		for (long i = 0; i < maxProcessingLoopIterations; ++i) {
+			store.advanceCurrentMinSecond(consensusTime);
+
+			List<ScheduleID> txnIdsToExpire = store.nextSchedulesToExpire(consensusTime);
```

Comment:
Is there a reason this variable is named `txnIdsToExpire` instead of e.g., `scheduleIdsToExpire`?